pkg/controller/clusterprovision/clusterprovision_controller.go

```diff
@@ -306,6 +324,28 @@ func (r *ReconcileClusterProvision) reconcileRunningJob(instance *hivev1.Cluster
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileClusterProvision) getInstallPod(job *batchv1.Job, pLog log.FieldLogger) (*corev1.Pod, error) {
+	installPod := &corev1.Pod{}
+	podList := &corev1.PodList{}
+	listOptions := &client.ListOptions{LabelSelector: labels.SelectorFromSet(map[string]string{"job-name": job.Name})}
```

Comment:
Use the selector from the job (`.spec.selector`) to identify the pod.

Reply:
Could not find any labels in` .spec.selector` that can identify pod
`spec:
  backoffLimit: 0
  completions: 1
  parallelism: 1
  selector:
    matchLabels:
      controller-uid: 0f5d9e8a-e57e-4d0c-ae86-a43ac3c0c637
`

Reply:
The `controller-uid: 0f5d9e8a-e57e-4d0c-ae86-a43ac3c0c637` labels is what you want. But more generically, the pod must match the selector, whatever it is. I am suggesting that you get the job, then get the pod based on the selector specified on the job.